kroxylicious-systemtests/src/main/java/io/kroxylicious/systemtests/resources/ResourceType.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.systemtests.resources;
+
+import io.fabric8.kubernetes.api.model.HasMetadata;
+
+/**
+ * The interface Resource type.
+ *
+ * @param <T>  the type parameter
+ */
+public interface ResourceType<T extends HasMetadata> {
```

Comment:
Intuitively it feels wrong that we'd need an interface that looks like this.  What's the motivation, why's fabric8 APIs not sufficient.

Reply:
I believe the motivation of that is to have control about what kind of resources can be or have been installed in kubernetes. This is the code inherited from Strimzi that manages the resources in the class ResourceManager

Reply:
I'm not really grasping what the abstraction is achieving. Perhaps the name isn't helping.  I don't think these are really "Types" at all.